src/main/java/org/opentripplanner/gtfs/mapping/RouteMapper.java

```diff
@@ -0,0 +1,46 @@
+package org.opentripplanner.gtfs.mapping;
+
+import org.opentripplanner.model.Route;
+import org.opentripplanner.util.MapUtils;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+class RouteMapper {
```

Comment:
Not all of these are entities, but I get the idea. I have used the hole scale from reflection generic mappers to this; these are very simple and easy to read - inline with level of abstractions in OTP. @abyrd Do you want me to fix this on top of PR #2584?

Reply:
I'm not advocating anything with reflection, I like the simple style. I just thought it might be nice to make it one notch more formal and give them a shallow hierarchy so instead of only the class names implying they have the same function, this shared purpose is reflected in a language construct (class hierarchy). If it's not too much trouble and we can agree that this is useful for people understanding the code in the future, then yes it would be good to include this change in #2584.